### ISSUE TITLE
peerAddress is required

### ISSUE THREAD

**Issue by michielbdejong** [ripple/ilp-plugin-xrp-paychan#20]
I'm trying to set up a server to which clients can connect using xrp-paychan. For the client, this works, but for the server, it says I also need to specify the peerAddress. Why is this, and can this be avoided?

**Comment by sharafian** [comment]
The topology that plugin XRP paychan supports right now is a bilateral setup with two unilateral payment channels. Because payment channels are established before any transfers can be sent either way on the channel, each plugin must know its peer address. Are you trying to make a system where many clients connect to one server?

**Comment by emschwartz** [comment]
> a system where many clients connect to one server

In order to have a connector that supports payment channels it either needs the server plugin to support multiple clients or it needs to spawn a server plugin for each user (that seems potentially problematic).

**Comment by sharafian** [comment]
Yeah, right now the only plugin we have that does many-to-one is the mini-accounts plugin. I suppose it would be possible to work off of that and add payment channel functionality, where it replaces the internal balance keeping with logic that validates the payment channels of connecting peers and makes sure their claims are valid.
